apps/che/src/main/fabric8/deployment.yml

```diff
@@ -144,13 +144,17 @@ spec:
         imagePullPolicy: "IfNotPresent"
         name: che
         readinessProbe:
-          tcpSocket:
+          httpGet:
+            path: /api/system/state
             port: 8080
-          initialDelaySeconds: 20
+            scheme: HTTP
+          initialDelaySeconds: 60
```

Comment:
@mlabuda I'm just trying to understand if we really need both readiness and liveness probes? Defining a livenessProbe `initialDelaySeconds` is enough to make it clear for openShift that it doesn't need to worry if che-server takes `initialDelaySeconds` to start. And yes without readiness probe we get 504 errors initially instead of 503s if we try to send and HTTP packet to the che-server. But does it really make a difference if, when starting che-server, we get a 503 (service unavailable) or a 504 (gateway timeout)? Maybe we need the readinessProbe because without it the che-starter would not know if che-server is up and running or not?

About optimisation I really think that we should all have a tatoo on our arm that remember us to: "**Make it run, make it right, make it fast**" 😃  (c.f. [Kent Beck](https://www.facebook.com/notes/kent-beck/mastering-programming/1184427814923414) )You can try to make it fast right away but you will most likely spend more time than trying to make it run first...